streaming/src/test/scala/org/apache/spark/streaming/ReceiverSuite.scala

```diff
@@ -129,6 +130,41 @@ class ReceiverSuite extends TestSuiteBase with Timeouts with Serializable {
     }
   }
 
+  test("kill receiver") {
+    val sparkConf = new SparkConf()
+      .setMaster("local[4]")
+      .setAppName(framework)
+    val batchDuration = Milliseconds(1000)
+    withStreamingContext(new StreamingContext(sparkConf, batchDuration)) { ssc =>
+      val receiver = new FakeReceiver
+      val executor = new FakeReceiverSupervisor(receiver)
+      val task = new FakeReceiverTask(0, executor)
+
+      new Thread("Receiver Task") {
+        override def run(): Unit = {
+          task.run(0, 0)
+        }
+      }.start()
+      Thread.sleep(2000)
+
+      // Verify that receiver was started
+      assert(receiver.onStartCalled)
+      assert(executor.isReceiverStarted)
+      assert(receiver.isStarted)
+      assert(!receiver.isStopped())
+      assert(receiver.otherThread.isAlive)
+
+      task.kill(false)
+
```

Comment:
There's some asynchrony in task killing, so these checks that the receiver was stopped need to be wrapped in an `eventually` block or some use some other mechanism to avoid race conditions.